### ISSUE TITLE
Build system improvements followup issues

### ISSUE THREAD

**Issue by wirew0rm** [fair-acc/opencmw-cpp#149]
After #142 was merged, there are some follow up improvements on the build system side which should be addressed in a followup PR:

- [ ] add a way to allow using system libraries (mostly only useful for zeromq, as the others are all header only)
  - relevant recent cmake improvements: [dependency providers](https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html#dependency-providers)
- [ ] fix the install directives, find out whether and where to install dependencies
- [x] fix clang build
- [x] other compilers, OSes, WASM
- [ ] static linking?
- [ ] reenable -Werror (commented out) needs warnings resolved first
- [x] disable gsl-lite warnings and bring `SYSTEM` includepath upstream in rxcpp
- [ ] Precompiled headers to decrease build time
- [ ] use gcovr from ubunut instead of pip? (as implemented already in graph-prototype)
